src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -187,14 +190,21 @@ public void InitializeService(ServiceHost serviceHost)
             ConnectionInfo connInfo;
             if (!ConnectionService.TryFindConnection(executeParams.OwnerUri, out connInfo))
```

Comment:
Minor: I was confused by use of `executeParams.OwnerUri` vs `executeStringParams.OwnerUri`. One is the original request URI, the other is a random one used to ensure we have a dedicated connection. Given this, I'd suggest using a variable called `randomUri` or `newUri` instead of `executeStringParams.OwnerUri`.